### PR TITLE
Add building instructions to the readme

### PR DESCRIPTION
**Pull request by dberliner**
@glfejer, could you read through this and let me know if its clear? David and I have been working too closely with this software to have much insight on its completeness.

### CODE REVIEW

**Review by drbild — APPROVED** [review]
LGTM, but agree that @glfejer should take a look.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,65 @@
 # Host Socket Sharing
 
-This repository contains the HSS specification as well as Xaptums host-side implementation. 
+This repository contains the HSS specification as well as Xaptums implementation for both hosts and devices.
```
Comment (glfejer) [inline README.md:3]: Xaptum's

File: README.md [hunk 2]
```diff
@@ -1,8 +1,65 @@
 # Host Socket Sharing
```
Comment (glfejer) [inline README.md:2]: Maybe add a sentence or two on what HSS is or how it works. Just a 10K ft view.

File: README.md [hunk 3]
```diff
@@ -1,8 +1,65 @@
...
+```
+git clone git@github.com:xaptum/hss.git
+cd hss/host/src
+make
+sudo make install
+```
+
+# HSS Device Drivers
+
+HSS requires out-of-tree modifications to the kernel of the USB device, as such patches must be applied to the kernel as it is build. Luckily, the patches are relatively simple. They are located in the `device/kernel` directory. The patch `hss-includes.patch` creates new includes to device HSS packet specifications within the kernel, `hss-socket.patch` creates a new type of socket `AF_HSS`, and `hss-selinux.patch` adds this new sockets definitions to SELinux's compile time checks. The second two patches may require some amount of customization to fit around a given version of Linux, but the addition is very simple.
```
Comment (glfejer) [inline README.md:50]: ... creates new `include` statements in the device HSS...

also - break up long lines.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,65 @@
...
+The `device/gadget` directory contains the USB driver responsible for communicating with the host.
+
+The `device/net` directory contains the Linux network layer driver responsible for handling socket communications, as well as software responsible for packetizing the data and communicating with the USB driver.
+
+Both loadable modules require a kernel with the included patches. The `f_hss` driver requires the `net` driver to be built first and compile-time access to its `Modules.symvers`.
+
+Xaptum maintains a Buildroot [project](https://github.com/xaptum/xaptum-buildroot) for our hardware that integrates HSS, this can be used as an example for integrating HSS into your project.
 
 ## License
 Copyright (c) 2018 Xaptum, Inc.
```
Comment (glfejer) [inline README.md:65]: -2020

**Review by glfejer — CHANGES_REQUESTED** [review]
some minor changes

**Pull request merged**